### PR TITLE
Integration Tests running in GH actions

### PR DESCRIPTION
**Pull request by davodesign84**
now rebased off of https://github.com/jet-lab/jet-v2/pull/19 thanks to @partiallysorted  for fully sorting the tests out :D
This now includes:
- working precommit linting
- working precommit standardised formatting with prettier
- working precommit  typescript compilation test
- working precommit anchor tests
- github actions pipeline for testing all of the above
- fixes for ~20/25 linting issues

### CODE REVIEW

File: .github/workflows/ts_coverage.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: build
```
[the remark this answers is not in the document]
Comment (davodesign84) [reply .github/workflows/ts_coverage.yml:1]: good shout, I just kept the naming convention from the existing GH action, but I do need to figure out how to get coverage from anchor, so perhaps I'll rename this for now

[the remark this answers is not in the document]
Comment (davodesign84) [reply .github/workflows/ts_coverage.yml:1]: ah right I only renamed it within the file, but forgot to rename the actual file :D Fix coming up

**Comment by Tristyn** [comment]
This is great

**Comment by davodesign84** [comment]
> This is great

Really it's all thanks to @partiallysorted for fixing the tests in the first place :)

**Comment by callensm** [comment]
can we combine the new workflow from the PR with the exist one that already uses the anchor and caching?

File: .github/workflows/ts_test.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+name: build
+
+env:
+  SOLANA_VERSION: 1.9.16
+  ANCHOR_CLI_VERSION: 0.23.0
+
+on:
+  push:
+    branches:
+      - 'master'
```
Comment (jetjake) [inline .github/workflows/ts_test.yml:10]: can we make this all branches?

Comment (callensm) [reply .github/workflows/ts_test.yml:10]: there also is a `pull_request` hook that will run the workflow every time a pr is created or updated which covers all incoming branches into `master`

**Comment by davodesign84** [comment]
> can we combine the new workflow from the PR with the exist one that already uses the anchor and caching?

Yup that's next on my list. But I know @jetjake wanted some form of test running for now

**Comment by davodesign84** [comment]
> can we combine the new workflow from the PR with the exist one that already uses the anchor and caching?

On a side note, what's the reasoning behind building on OSX?

**Comment by callensm** [comment]
> > can we combine the new workflow from the PR with the exist one that already uses the anchor and caching?
> 
> On a side note, what's the reasoning behind building on OSX?

what is building on osx?

**Review by callensm — APPROVED** [review]

**Review by callensm — APPROVED** [review]

**Pull request merged**